pkg/lib/repo/local/repository.go

```diff
@@ -0,0 +1,197 @@
+package local
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"sort"
+
+	"kitops/pkg/lib/constants"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/errdef"
+)
+
+type localIndex struct {
+	indexPath string
+	modelTags *tagsIndex
+	ocispec.Index
+}
+
+func newLocalIndex(storagePath, repoName string) (*localIndex, error) {
+	li := &localIndex{}
+	indexPath := constants.IndexJsonPathForRepo(storagePath, repoName)
+	index, err := parseIndex(indexPath)
+	if err != nil {
+		return nil, err
+	}
+	li.indexPath = indexPath
+	li.Index = *index
+
+	tagsIndexPath := constants.TagIndexPathForRepo(storagePath, repoName)
+	tags, err := parseTagsIndex(tagsIndexPath)
+	if err != nil {
+		return nil, err
+	}
+	li.modelTags = tags
+
+	return li, nil
+}
+
+func (li *localIndex) addManifest(manifestDesc ocispec.Descriptor) error {
+	curTag := manifestDesc.Annotations[ocispec.AnnotationRefName]
+	delete(manifestDesc.Annotations, ocispec.AnnotationRefName)
+	li.Manifests = append(li.Manifests, manifestDesc)
+	if err := li.save(); err != nil {
+		return err
+	}
+	if curTag != "" {
+		li.modelTags.tagToDigest[curTag] = manifestDesc
+		if err := li.modelTags.save(); err != nil {
```

Comment:
Yeah, makes sense. I think I had it like this because it's annoying to have to refactor the lines to add some context to the error, but it's cleaner your way.